flame/data.py

```
@@ -381,57 +380,134 @@
+                    )
+
+                # --- cu_seqlens validation checks remain the same ---
+                if batch['cu_seqlens'].numel() < 2:
+                     raise ValueError(f"Calculated cu_seqlens must have at least start and end: {batch['cu_seqlens']}")
+                if not torch.all(batch['cu_seqlens'][1:] >= batch['cu_seqlens'][:-1]):
+                     raise ValueError(f"Calculated cu_seqlens are not monotonically increasing: {batch['cu_seqlens']}")
+                if batch['cu_seqlens'][0] != 0:
+                    raise ValueError(f"Calculated cu_seqlens do not start at 0: {batch['cu_seqlens']}")
+                if batch['cu_seqlens'][-1] != batch['input_ids'].size(1):
+                     # Allow empty sequence case where cu_seqlens=[0, 0] and input_ids.size(1)=0
+                    if not (batch['cu_seqlens'].tolist() == [0, 0] and batch['input_ids'].size(1) == 0):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)